kiele

```diff
@@ -12,11 +12,11 @@ export PATH="$INSTALL_BIN:$INSTALL_LIB:$PATH"
 export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:-}:/usr/local/lib
 
 check_k_install() {
-    which kore-exec &> /dev/null \
+    type -p kore-exec &> /dev/null \
         || fatal "Must have K installed! See https://github.com/kframework/k/releases."
-    which kast &> /dev/null \
+    type -p kast &> /dev/null \
         || fatal "Must have K installed! See https://github.com/kframework/k/releases."
-    which krun &> /dev/null \
+    type -p krun &> /dev/null \
         || fatal "Must have K installed! See https://github.com/kframework/k/releases."
```

Comment:
What's the reasoning behind using `type` instead of `which`? Because maybe `which` is not installed and if they have `bash` then they have `type`?

Reply:
There's not really any notion of "installed" with Nix. I can make `kiele` depend on `which`, but `type` is built-in to Bash and does the same thing in this case. This is a standard tweak that we often have to make to scripts that are being packaged with Nix, but I have no problem pulling in the `which` package if you prefer.